ecs-agent/netlib/data/network_client.go

```diff
@@ -14,23 +14,61 @@
 package data
 
 import (
+	"crypto/rand"
 	"encoding/json"
+	"math/big"
+	"strconv"
 
 	"github.com/aws/amazon-ecs-agent/ecs-agent/data"
 	"github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
 	"github.com/aws/amazon-ecs-agent/ecs-agent/netlib/model/tasknetworkconfig"
 
+	"github.com/pkg/errors"
 	bolt "go.etcd.io/bbolt"
 )
 
 const (
 	networkNamespaceBucketName = "networkNamespaceConfiguration"
+	portBucketName             = "port"
+
+	// geneveDstPortMin and geneveDstPortMax specifies the range from which the GENEVE destination port will be selected.
+	// The range is currently set from 6081 to 6151 which gives us 70 ports per baremetal host to access the same subnet.
+	// The goal while choosing these values were to keep the range wide enough so as to avoid port exhaustion and narrow
+	// enough to reduce the attack surface since the same port range has to be opened up in the security group
+	// assigned to the connectivity provider ENI. A rough calculation that explains why 70 ports were chosen as the range is:
+	// Our largest customers have been seen to launch somewhat about 10,000 tasks around a given time. Ideally there
+	// are 3 AZs per region with 2 cells. Each cell has 8 instances. We will also establish tunnels between the target
+	// subnet and at least 3 random connectivity providers per cell. This means we'll be able to spread customer
+	// tasks across at least 16 hosts and across 6 tunnels per AZ. Assuming even distribution, then this puts us
+	// at 69 uVMS per host accessing the same subnet (10000 / [6 * 8 * 3]).
+	// We start with 6081 since that is the default destination port for GENEVE interfaces.
+	geneveDstPortMin = 6081
+	geneveDstPortMax = 6151
+
+	// geneveDstPortDefault is the default port to be assigned if not taken already.
+	geneveDstPortDefault = 6081
```

Comment:
Should we do `geneveDstPortDefault = geneveDstPortMin` instead since we mentioned that 6081 is the default value AND the value that we start the range with?

Reply:
I have removed implementation of NetworkDataClient from the shared library because the agent is able to inject the same into the network builder.